controllers/appmesh/custom/custom_controller.go

```diff
@@ -0,0 +1,244 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package custom
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+)
+
+const (
+	// API Server QPS
+	DefaultAPIServerQPS   = 10
+	DefaultAPIServerBurst = 15
+)
+
+// Converter for converting k8s object and object list used in watches and list operation
+type Converter interface {
+	// ConvertObject takes an object and returns the modified object which will be
+	// stored in the data store
+	ConvertObject(originalObj interface{}) (convertedObj interface{}, err error)
+	// ConvertList takes an object and returns the modified list of objects which
+	// will be returned to the Simple Pager function to aggregate the list pagination
+	// response
+	ConvertList(originalList interface{}) (convertedList interface{}, err error)
+	// Resource returns the K8s resource name to list/watch
+	Resource() string
+	// ResourceType returns the k8s object to list/watch
+	ResourceType() runtime.Object
+}
+
+// Controller Interface implemented by NewController
+type Controller interface {
+	// StartController starts the controller. Will block the calling routine
+	StartController(dataStore cache.Indexer, stopChanel chan struct{})
+	// GetDataStore returns the data store once it has synced with the API Server
+	GetDataStore() cache.Indexer
+}
+
+// NewController is an Informer which converts Pod Objects and notifies corresponding even handlers via Channels
+type NewController struct {
+	// ClientSet is the kubernetes client set
+	ClientSet *kubernetes.Clientset
```

Comment:
let's use lowercase for private members like `clientSet`

Reply:
Its used in main.go while initializing the struct. So only the exposed variables are Uppercase